quartz/processors/parse.ts

```diff
@@ -24,14 +26,19 @@ export function createProcessor(ctx: BuildCtx): QuartzProcessor {
       .use(remarkParse)
       // MD AST -> MD AST transforms
       .use(
-        transformers
-          .filter((p) => p.markdownPlugins)
-          .flatMap((plugin) => plugin.markdownPlugins!(ctx)),
-      )
+        transformers.flatMap((plugin) => plugin.markdownPlugins?.(ctx) ?? []),
```

Comment:
this is just wrong. what is the problem with previous logics? we filter whether a plugin has markdownPlugins then flatmap it?

Reply:
> this is just wrong

No?. It does literally the same thing, just without the `!`, and a bit shorter.
Combining filter and flatMap by using empty lists is a classic.

Since I was touching the parse anyway I also did this, I didn't even really think much about it, sorry :upside_down_face:

But yeah this is just bikeshedding, I can undo it if you hate it.

Reply:
Let's just keep the changes related to this PR.